impl/src/main/java/com/okta/sdk/impl/oauth2/AccessTokenRetrieverServiceImpl.java

```diff
@@ -128,23 +127,26 @@ public OAuth2AccessToken getOAuth2AccessToken() throws IOException, InvalidKeyEx
      * @return signed JWT string
      * @throws InvalidKeyException
      * @throws IOException
+     * @throws IllegalArgumentException if <code>tokenTtl</code> property is missing or negative
      */
     String createSignedJWT() throws InvalidKeyException, IOException {
+        long tokenTtl = tokenClientConfiguration.getAccessTokenTtl().getSeconds();
+        if (tokenTtl < 1) {
+            throw new IllegalArgumentException("'okta.client.accessTokenTtl' property is missing or negative");
+        }
         String clientId = tokenClientConfiguration.getClientId();
         PrivateKey privateKey = parsePrivateKey(getPemReader());
         Instant now = Instant.now();
 
-        String jwt = Jwts.builder()
+        return Jwts.builder()
             .setAudience(tokenClientConfiguration.getBaseUrl() + TOKEN_URI)
             .setIssuedAt(Date.from(now))
-            .setExpiration(Date.from(now.plus(1L, ChronoUnit.HOURS)))
+            .setExpiration(Date.from(now.plusSeconds(tokenTtl)))
```

Comment:
nit: if accessTokenTtl is a duration, you should be able to call `now.plus(tokenTtl)` or something like that

Reply:
You are always right 👍